tests/percy/iframephotographer.js

```diff
@@ -0,0 +1,55 @@
+const Photographer = require('./photographer');
+const { waitTillHTMLRendered } = require('./utils');
+const { SnapshotWidths } = require('./constants');
+
+/**
+ * Responsible for taking Percy snapshots of an iframe answers experience
+ */
+class IframePhotographer extends Photographer {
+  /**
+   * @param {Object} obj
+   * @param {import('puppeteer').Page} obj.page A Pupeteer Page
+   * @param {function} obj.percySnapshot The percy snapshot function
+   * @param {string} obj.siteUrl A url to the index of the site
+   */
+  constructor({ page, percySnapshot, siteUrl, iframePage }) {
+    super();
+    this._page = page;
+    this._percySnapshot = percySnapshot;
+    this._siteUrl = siteUrl;
+    this._iframePage = iframePage;
+  }
+
+  async gotoUniversalPage({ queryParams = '' } = {}) {
+    const url = `${this._siteUrl}/${this._iframePage}?${queryParams}`;
+    await this._page.goto(url);
+    await waitTillHTMLRendered(this._page);
+  }
+
+  async gotoVerticalPage({ vertical, queryParams = '' } = {}) {
+    const url = `${this._siteUrl}/${this._iframePage}?verticalUrl=${vertical}&${queryParams}`;
+    await this._page.goto(url);
+    await waitTillHTMLRendered(this._page);
+  }
+
+  async snapshot(snapshotName) {
+    await this._percySnapshot(snapshotName + '-iframe');
```

Comment:
instead of separating everything with just dashes, would we want to try something similar to BEM? it looks like the different dials we have for snapshots are page template, width, and then description? It's unfortunate that percy doesn't let us have organization beyond just a string

Reply:
For BEM, are you think that we just use two dashes? Or something else?

Reply:
I was thinking something like `verticalFullPageMap--mobile__listView` or `verticalStandard__search`? It could be worth another item to think about this a little more, but I kind of like making some improvements right now since we're adding in the -iframe suffix.

Reply:
maybe we even have 4 different categories like `pageTemplate--width__description-integrationType`?

Reply:
What do you think of putting the layout next to the integratinType? Like: `pageTemplate__description-layout-integrationType` For example, that could look like: `vertical-full-page-map__nlp-filters-desktop-view-iframe`

Reply:
that's fine by me! I would just change it so that description, layout, and integration type aren't also separated by dashes.
maybe `vertical-full-page-map--nlp-filters__desktop-view__iframe`?

Reply:
Sure, that works! And maybe a single underscore before `iframe`?

Reply:
yeah makes sense to me!